cadquery/cq.py

```diff
@@ -2728,6 +2728,66 @@ def _sweep(self, path, multisection=False, makeSolid=True, isFrenet=False,
 
         return Compound.makeCompound(toFuse)
 
+    def interpPlate(self, surf_edges, surf_pts=[], thickness=0, combine=True, clean=True, Degree=3, NbPtsOnCur=15, NbIter=2, Anisotropie=False, Tol2d=0.00001, Tol3d=0.0001, TolAng=0.01, TolCurv=0.1, MaxDeg=8, MaxSegments=9):
+        """
+        Returns a plate surface that is 'thickness' thick, enclosed by 'surf_edge_pts' points,  and going through 'surf_pts' points.
+        
+        :param surf_edges
+        :type 1 surf_edges: list of [x,y,z] float ordered coordinates
+        :type 2 surf_edges: list of unordered CadQuery wires
+        :param surf_pts = [] (uses only edges if [])
+        :type surf_pts: list of [x,y,z] float coordinates
+        :param thickness = 0 (returns 2D surface if 0)
+        :type thickness: float >= 0
+        :param combine: should the results be combined with other solids on the stack
+            (and each other)?
+        :type combine: true to combine shapes, false otherwise.
+        :param boolean clean: call :py:meth:`clean` afterwards to have a clean shape      
+        :param Degree = 3 (OCCT default)
+        :type Degree: Integer >= 2
+        :param NbPtsOnCur = 15 (OCCT default)
+        :type: NbPtsOnCur Integer >= 15
+        :param NbIter = 2 (OCCT default)
+        :type: NbIterInteger >= 2
+        :param Anisotropie = False (OCCT default)
+        :type Anisotropie: Boolean
+        :param: Tol2d = 0.00001 (OCCT default)
+        :type Tol2d: float > 0
+        :param Tol3d = 0.0001 (OCCT default)
+        :type Tol3dReal: float > 0
+        :param TolAng = 0.01 (OCCT default)
+        :type TolAngReal: float > 0
+        :param TolCurv = 0.1 (OCCT default)
+        :type TolCurvReal: float > 0
+        :param MaxDeg = 8 (OCCT default)
+        :type MaxDegInteger: Integer >= 2 (?)
+        :param MaxSegments = 9 (OCCT default)
+        :type MaxSegments: Integer >= 2 (?)
+        """
+        
+        # If thickness is 0, only a 2D surface will be returned.
+        if thickness==0:
+            combine=False
+        
+        # If a list of wires is provided, make a closed wire
+        if not isinstance(surf_edges, list): 
+            surf_edges = [o.vals()[0] for o in surf_edges.all()]
+            surf_edges = Wire.assembleEdges(surf_edges)
+            surf_edges = surf_edges.wrapped
```

Comment:
I think it would be more elegant to do this on the interpPlate (i.e. occ_impl) level

Reply:
Done. Yes it is cleaner and actually saved 3 lines of code.